src/Components/Server/src/DependencyInjection/ConfigureStaticFilesOptions.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.Server.DependencyInjection
+{
+    public class ConfigureStaticFilesOptions : IPostConfigureOptions<StaticFileOptions>
+    {
+        public ConfigureStaticFilesOptions(IWebHostEnvironment environment)
+        {
+            Environment = environment;
+        }
+
+        public IWebHostEnvironment Environment { get; }
+
+        public void PostConfigure(string name, StaticFileOptions options)
+        {
+            name = name ?? throw new ArgumentNullException(nameof(name));
+            options = options ?? throw new ArgumentNullException(nameof(options));
+
+            if (name != Options.DefaultName)
+            {
+                return;
+            }
+
+            // Basic initialization in case the options weren't initialized by any other component
+            options.ContentTypeProvider = options.ContentTypeProvider ?? new FileExtensionContentTypeProvider();
+            if (options.FileProvider == null && Environment.WebRootFileProvider == null)
+            {
+                throw new InvalidOperationException("Missing FileProvider.");
+            }
+
+            options.FileProvider = options.FileProvider ?? Environment.WebRootFileProvider;
+
+            var prepareResponse = options.OnPrepareResponse;
+            if (prepareResponse == null)
+            {
+                options.OnPrepareResponse = BlazorApplicationBuilderExtensions.SetCacheHeaders;
+            }
+            else
+            {
+                void PrepareResponse(StaticFileResponseContext context)
+                {
+                    prepareResponse(context);
+                    BlazorApplicationBuilderExtensions.SetCacheHeaders(context);
+                }
+
+                options.OnPrepareResponse = PrepareResponse;
+            }
+
+            // Add our provider
+            var provider = new ManifestEmbeddedFileProvider(typeof(ConfigureStaticFilesOptions).Assembly);
+
+            options.FileProvider = new CompositeFileProvider(provider, new ContentReferencesFileProvider(), options.FileProvider);
+        }
+
+        private class ContentReferencesFileProvider : IFileProvider
+        {
+            byte[] _data = Encoding.UTF8.GetBytes(@"{ ""cssReferences"": [], ""jsReferences"": [] }");
```

Comment:
unused. You kinda teased me there for a minute.

Reply:
https://github.com/aspnet/AspNetCore/pull/7770/files#diff-ff067b68eddd5630473748d5d100f403R77

Reply:
Yeah, we could pull this stuff out completely, but I'd be inclined to leave things as-is until we spec out and implement static-file-from-nuget-package support properly. No need to reshuffle things multiple times.

Reply:
Agree